### PR TITLE
fixes #8892, #9340 - Validates content label and override value, BZ1173723, BZ1187750

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

File: app/controllers/katello/api/v2/activation_keys_controller.rb [hunk 1]
```diff
@@ -206,23 +206,28 @@ def remove_subscriptions
 
     api :PUT, "/activation_keys/:id/content_override", N_("Override content for activation_key")
     param :id, :identifier, :desc => N_("ID of the activation key"), :required => true
-    param :label, String, :desc => N_("Label of the content"), :required => false
-    param :value, :number, :desc => N_("Override value 0/1"), :required => false
+    param :content_override, Hash, :desc => N_("Content override parameters") do
+      param :content_label, String, :desc => N_("Label of the content"), :required => true
+      param :value, :bool, :desc => N_("Override to 0/1, or 'default'"), :required => true
+    end
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: would this api change affect hammer at all?  what was the reasoning behind making the api change (versus keeping the parameters the same)?

Comment (cfouant) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: I actually didn't realize hammer could handle nested params. I've tested it in hammer and it seems to work, and removing the different handling of the two ways to handle cleans the code up a bit.

**Comment by cfouant** [comment]
@jlsherrill - Hammer is able to handle the nested params, which I was unaware of previously. So to make everything more uniform, I removed the different handling of the calls between CLI, API and UI. I also updated the UI to keep everything handled more uniformly as well (call in 'default' rather than null).

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**